### PR TITLE
Add set discard superseded method to api

### PR DESCRIPTION
**Pull request by matherno**
Added proj_operation_factory_context_set_discard_superseded() function to API.
Also made it so that FilterResults::removeUninterestingOps() is not called if not discarding superseded operations.

### CODE REVIEW

**Comment by kbevers** [comment]
Please update `scripts/reference_exported_symbols.txt`:

> Difference(s) found in exported symbols. If intended, refresh scripts/reference_exported_symbols.txt with 'scripts/dump_exported_symbols.sh src/.libs/libproj.so > scripts/reference_exported_symbols.txt'

**Comment by kbevers** [comment]
Closes https://github.com/OSGeo/PROJ/issues/1519

File: scripts/reference_exported_symbols.txt [hunk 1]
```diff
@@ -947,6 +947,7 @@ proj_operation_factory_context_set_desired_accuracy
 proj_operation_factory_context_set_grid_availability_use
 proj_operation_factory_context_set_spatial_criterion
 proj_operation_factory_context_set_use_proj_alternative_grid_names
+proj_operation_factory_context_set_discard_superseded
```
Comment (rouault) [inline scripts/reference_exported_symbols.txt]: symbols should be sorted. so move this line just after proj_operation_factory_context_set_desired_accuracy

**Comment by rouault** [comment]
Could you also please add a test in test/unit/test_c_api.cpp ?

**Comment by rouault** [comment]
@matherno ping ^

**Comment by kbevers** [comment]
@matherno Thanks for your contribution. Sorry for not getting round to merging this sooner!

**Pull request merged**